src/github.ts

```diff
@@ -257,16 +264,23 @@ export const backportPrExists = async (
     { headers: HEADERS },
   );
   const json = await response.json();
-  if (json.total_count > 0) return true;
+  if (json.total_count > 0) {
+    backportPrExistsCache.add(cacheKey);
+    return true;
+  }
 
   // also check if a branch that looks like the backport branch (getPrBranchName) exists
   response = await fetch(
     `${GITHUB_API}/repos/${Deno.env.get("BACKPORTER_GITEA_FORK")}/branches/${
       getPrBranchName(pr.number, giteaMajorMinorVersion)
     }`,
-    { headers: HEADERS },
+    { headers: HEADERS, method: "HEAD" },
```

Comment:
HEAD method counts the same as GET for [the rate limit](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#calculating-points-for-the-secondary-rate-limit), but it's still a good change to keep.